src-java/kilda-model/src/main/java/org/openkilda/model/ServiceCookie.java

```diff
@@ -63,7 +65,18 @@ public void validate() throws InvalidCookieException {
 
     // TODO - replace existing meter-id generation approach
     public MeterId getMeterId() {
-        return new MeterId(getServiceTag().getValue());
+        return new MeterId(getField(SERVICE_TAG_FIELD));
+    }
+
+    /**
+     * Extract and return "service tag" field is save way (return empty {@link Optional} object if tag is invalid).
+     */
+    public Optional<ServiceCookieTag> getServiceTagSafe() {
```

Comment:
I don't like idea having Safe method here, why not to change existing one?

Reply:
It can be done only if we don't want simple/raw access.